documentation/api/addType.md

```diff
@@ -45,6 +45,15 @@ Optional members:
 Adding new types to the system is best explained by an example. Let's
 say we wanted to add first class support for a `Person` type:
 
+<!-- unexpected-markdown hide:true -->
+
+```js
+const originalExpect = expect;
+expect = expect.clone();
+```
+
+<!-- unexpected-markdown persist:true -->
```

Comment:
Is this a leftover from before `freshContext` was introduced?

Reply:
No - it's related to that extra comment about `freshContext` :) Inside of such a context, you have a completely fresh global (not just a fresh expect because we're attempting to make something generic). If you need something extra to exist inside a fresh context, you have to mark it as such - in this case, the document as written needs `Person` to exist inside the fresh context.

Reply:
Oh, I see. Could that be the default?

Reply:
The bit I’m struggling with is then why does freshContext mean: to my mind it means a fresh global with anything else that’s been specified re-required. If everything you’ve already defined if present in a freshContext, then isn’t it somewhat undermined?

Reply:
Ah, now I understand. That is a bit of a power feature :)

I would also be cool with just repeating the `Person` constructor after each `freshContext`:

```diff
diff --git a/documentation/api/addType.md b/documentation/api/addType.md
index 9f457692..a5a3e30a 100644
--- a/documentation/api/addType.md
+++ b/documentation/api/addType.md
@@ -52,21 +52,17 @@ const originalExpect = expect;
 expect = expect.clone();
 ```
 
-<!-- unexpected-markdown persist:true -->
-
-```js
-function Person(name, age) {
-  this.name = name;
-  this.age = age;
-}
-```
-
 We start out by creating a basic type for handling `Person`
 instances. The name of the type should be `Person` and it should
 inherit from the built-in `object` type. Furthermore we add an
 `identify` method that will recognize `Person` instances.
 
 ```js
+function Person(name, age) {
+  this.name = name;
+  this.age = age;
+}
+
 expect.addType({
   name: 'Person',
   base: 'object',
@@ -106,6 +102,11 @@ constructor. We can fix that by implementing an `inspect` method on the type.
 <!-- unexpected-markdown freshContext:true -->
 
 ```js
+function Person(name, age) {
+  this.name = name;
+  this.age = age;
+}
+
 expect.addType({
   name: 'Person',
   base: 'object',
@@ -160,6 +161,11 @@ age. Then we need to override the `equal` method:
 <!-- unexpected-markdown freshContext:true -->
 
 ```js
+function Person(name, age) {
+  this.name = name;
+  this.age = age;
+}
+
 expect.addType({
   name: 'Person',
   base: 'object',
@@ -187,6 +193,11 @@ following way:
 <!-- unexpected-markdown freshContext:true -->
 
 ```js
+function Person(name, age) {
+  this.name = name;
+  this.age = age;
+}
+
 expect.addType({
   name: 'Person',
   base: 'object',
```

Reply:
https://github.com/unexpectedjs/unexpected/pull/721/commits/7606bba7b18d81f18dc9669185d8ec9a39b4a5f5